sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,2050 @@
+/// Tag for the 'C2PA' table.
+const C2PA_TABLE_TAG: TableTag = TableTag { data: *b"C2PA", };
+/// Tag for the 'head' table in a font.
+const HEAD_TABLE_TAG: TableTag = TableTag { data: *b"head", };
+
+/// 32-bit font-format identifier.
+///
+/// Embedded OpenType and MicroType Express formats cannot be detected with a
+/// simple magic-number sniff. Conceivably, EOT could be dealt with as a
+/// variation on SFNT, but MTX needs more exotic handling.
+enum Magic {
+    /// 'OTTO' - OpenType
+    OpenType = 0x4f54544f,
+    /// FIXED 1.0 - TrueType (or possibly v1.0 Embedded OpenType)
+    TrueType = 0x00010000,
+    /// 'wOFF' - WOFF 1.0
+    Woff = 0x774f4646,
+    /// 'wOF2' - WOFF 2.0
+    Woff2 = 0x774f4632,
+}
+
+/// Used to attempt conversion from u32 to a Magic value.
+impl TryFrom<u32> for Magic {
+    type Error = ();
+
+    /// Tries to convert from u32 to a valid font version.
+    fn try_from(v: u32) -> core::result::Result<Self, Self::Error> {
+        match v {
+            x if x == Magic::TrueType as u32 => Ok(Magic::TrueType),
+            x if x == Magic::OpenType as u32 => Ok(Magic::OpenType),
+            x if x == Magic::Woff as u32 => Ok(Magic::Woff),
+            x if x == Magic::Woff2 as u32 => Ok(Magic::Woff2),
+            _ => Err(()),
+        }
+    }
+}
+
+/// 'C2PA' font table - in storage
+#[derive(Debug)]
+#[repr(C, packed)]       // As defined by the C2PA spec.
+#[allow(non_snake_case)] // As named by the C2PA spec.
+struct TableC2PARaw {
+    majorVersion: u16,
+    minorVersion: u16,
+    activeManifestUriOffset: u32,
+    activeManifestUriLength: u16,
+    manifestStoreOffset: u32,
+    manifestStoreLength: u32,
+}
+impl TableC2PARaw {
+    fn write<TDest: Write + ?Sized>(&mut self, destination: &mut TDest) -> Result<()> {
+        destination.write_u16::<BigEndian>(self.majorVersion)?;
+        destination.write_u16::<BigEndian>(self.minorVersion)?;
+        destination.write_u32::<BigEndian>(self.activeManifestUriOffset)?;
+        destination.write_u16::<BigEndian>(self.activeManifestUriLength as u16)?;
+        destination.write_u32::<BigEndian>(self.manifestStoreOffset)?;
+        destination.write_u32::<BigEndian>(self.manifestStoreLength)?;
+        Ok(())
+    }
+}
+
+/// 'C2PA' font table - after loading from storage
+#[derive(Clone, Debug)]
+struct TableC2PA {
+    /// Major version of the C2PA table record
+    major_version: u16,
+    /// Minor version of the C2PA table record
+    minor_version: u16,
+    /// Optional URI to an active manifest
+    active_manifest_uri: Option<String>,
+    /// Optional embedded manifest store
+    manifest_store: Option<Vec<u8>>,
+}
+
+impl TableC2PA {
+    /// Creates a new C2PA record with the current default version information.
+    pub fn new(active_manifest_uri: Option<String>, manifest_store: Option<Vec<u8>>) -> Self {
+        Self {
+            active_manifest_uri: active_manifest_uri,
```

Comment:
Since they have the same name, can replace `active_manifest_uri: active_manifest_uri` with `active_manifest_uri`.

Reply:
Same for the rest of the fields as well.